app/code/Magento/Catalog/Block/Product/View.php

```diff
@@ -88,6 +94,7 @@ public function __construct(
         \Magento\Customer\Model\Session $customerSession,
         ProductRepositoryInterface $productRepository,
         \Magento\Framework\Pricing\PriceCurrencyInterface $priceCurrency,
+        \Magento\Catalog\Model\ProductFactory $productFactory,
```

Comment:
@maghamed I will make the changes tomorrow. But in general you are ok with the changes as they are presented now?